IPython/core/ultratb.py

```diff
@@ -248,6 +248,7 @@ def fix_frame_records_filenames(records):
         # Look inside the frame's globals dictionary for __file__,
         # which should be better. However, keep Cython filenames since
         # we prefer the source filenames over the compiled .so file.
+        filename = py3compat.cast_unicode(filename, "utf-8")
```

Comment:
I suspect that all of these can probably be `cast_unicode_py2` - the problem doesn't occur on Python 3, does it?